tools/java/org/unicode/cldr/util/VettingViewer.java

```diff
@@ -805,6 +807,161 @@ public void generateSummaryHtmlErrorTables(Appendable output, EnumSet<Choice> ch
 
     }
 
+    private class WriteContext {
+
+        private List<String> names = new ArrayList<>();
+        private List<String> locales = new ArrayList<>();
+        private StringBuffer[] outputs;
+        private EnumSet<Choice> choices;
+        private EnumSet<Choice> thingsThatRequireOldFile;
+        private EnumSet<Choice> ourChoicesThatRequireOldFile;
+        private T organization;
+        private VettingViewer<T>.FileInfo totals;
+        private Map<String, VettingViewer<T>.FileInfo> localeNameToFileInfo;
+        private String header;
+        private int configParallel; // parallelism. 0 means "let Java decide"
+        private int configChunkSize; // Number of locales to process at once, minimum 1
+
+        @SuppressWarnings("unchecked")
+        public WriteContext(Set<Entry<String, String>> entrySet, EnumSet<Choice> choices, T organization, FileInfo totals, Map<String, VettingViewer<T>.FileInfo> localeNameToFileInfo, String header) {
+            for(Entry<String, String> e : entrySet) {
+                names.add(e.getKey());
+                locales.add(e.getValue());
+            }
+            int count = names.size();
+            this.outputs = new StringBuffer[count];
+            for(int i=0;i<count;i++) {
+                outputs[i] = new StringBuffer();
+            }
+            if(DEBUG_THREADS) System.err.println("Initted " + this.outputs.length + " outputs");
+
+            // other data
+            this.choices = choices;
+
+            thingsThatRequireOldFile = EnumSet.of(Choice.englishChanged, Choice.missingCoverage, Choice.changedOldValue);
+            ourChoicesThatRequireOldFile = choices.clone();
+            ourChoicesThatRequireOldFile.retainAll(thingsThatRequireOldFile);
+
+            this.organization = organization;
+            this.totals = totals;
+            this.localeNameToFileInfo = localeNameToFileInfo;
+            this.header = header;
+
+            if(DEBUG_THREADS) System.err.println("writeContext for " + organization.toString() + " booted with " + count + " locales");
+
+            // setup env
+            CLDRConfig config = CLDRConfig.getInstance();
+
+            this.configParallel = Math.max(config.get("CLDR_VETTINGVIEWER_PARALLEL", 0), 0);
+            if(this.configParallel < 1) {
+                this.configParallel = java.lang.Runtime.getRuntime().availableProcessors(); // matches ForkJoinPool() behavior
+            }
+            this.configChunkSize = Math.max(config.get("CLDR_VETTINGVIEWER_CHUNKSIZE", 1), 1);
+            System.err.println("vv: CLDR_VETTINGVIEWER_PARALLEL="+configParallel+", CLDR_VETTINGVIEWER_CHUNKSIZE="+configChunkSize);
+        }
+
+        public void append(Appendable output) throws IOException {
```

Comment:
This name is misleading. It is appending _this_ to output, rather than the reverse, so should be _appendto_.